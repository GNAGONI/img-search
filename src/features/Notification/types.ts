enum NotificationKind { 
  SUCCESS = 'success',
  INFO = 'info',
  ERROR = 'error',
  WARNING = 'warning',
  NULL = ''
}

type NotificationData = { 
  message: string;
}

type NotificationState = {
  open: boolean;
  kind: NotificationKind;
} & NotificationData;

export { NotificationKind };
export type {
  NotificationState,
  NotificationData,
};
