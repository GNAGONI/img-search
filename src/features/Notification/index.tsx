import Stack from '@mui/material/Stack';
import Snackbar from '@mui/material/Snackbar';
import Alert from '@mui/material/Alert';
import { useSelector } from 'react-redux';
import { getNotification } from './state/selectors';
import { hideNotification } from './state/slice';
import useAppDispatch from '../../hooks/useAppDispatch';

const Notification = () => {
  const { message, kind, open } = useSelector(getNotification);
  const dispatch = useAppDispatch();

  const handleClose = () => {
    dispatch(hideNotification());
  };

  return (
    <Stack spacing={2} sx={{ width: '100%' }}>
      <Snackbar open={open} autoHideDuration={4000} onClose={handleClose}>
        <Alert onClose={handleClose} severity={kind || 'info'} sx={{ width: '100%' }}>
          {message || ''}
        </Alert>
      </Snackbar>
    </Stack>
  );
};

export default Notification;
