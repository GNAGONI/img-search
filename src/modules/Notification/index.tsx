import Stack from '@mui/material/Stack';
import Snackbar from '@mui/material/Snackbar';
import Alert from '@mui/material/Alert';
import { useSelector, useDispatch } from 'react-redux';
import { getNotification } from './state/selectors';
import { hideNotification } from './state/slice';

const Notification = () => {
  const { message, kind, open } = useSelector(getNotification);
  const dispatch = useDispatch();

  const handleClose = () => {
    dispatch(hideNotification());
  };

  return (
    <Stack spacing={2} sx={{ width: '100%' }}>
      <Snackbar open={open} autoHideDuration={4000} onClose={handleClose}>
        <Alert onClose={handleClose} severity={kind || 'info'} sx={{ width: '100%' }}>
          {message || ''}
        </Alert>
      </Snackbar>
    </Stack>
  );
};

export default Notification;
