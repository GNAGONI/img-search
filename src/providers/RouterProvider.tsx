import { BrowserRouter, Routes, Route } from "react-router-dom";
import Layout from "../features/Layout";
import Login from "../modules/Login";
import Home from "../modules/Home";
import Error from "../components/Error";
import PrivateRoute from '../features/PrivateRoute';

const RouterProvider = () => (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<Layout />}>
        <Route index element={<Login />} />
        <Route
          path="/home"
          element={(
            <PrivateRoute>
              <Home />
            </PrivateRoute>
          )}
        />
        <Route path="*" element={<Error />} />
      </Route>
    </Routes>
  </BrowserRouter>
);

export default RouterProvider;
