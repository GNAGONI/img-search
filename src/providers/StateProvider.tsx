import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import { all } from 'redux-saga/effects';
import userSagas from '../components/Login/state/sagas';
import userReducer from '../components/Login/state/slice';
import homeSagas from '../components/Home/state/sagas';
import homeReducer from '../components/Home/state/slice';


function* rootSaga() {
  yield all([
    ...userSagas,
    ...homeSagas,
  ]);
}

const sagaMiddleware = createSagaMiddleware();

type Props = {
  children: ReactElement,
}

const setupStore = () => {
  const store = configureStore({
    reducer: {
      user: userReducer,
      home: homeReducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware({ thunk: false })
      .concat(sagaMiddleware) // TODO: check for prepend
      .concat(logger),
  });
  return store;
}

const store = setupStore();

const StateProvider = ({ children }: Props) => {
  sagaMiddleware.run(rootSaga);

  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
};

export default StateProvider;
