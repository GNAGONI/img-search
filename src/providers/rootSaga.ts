import { all } from 'redux-saga/effects';
import userSagas from '../components/Login/state/sagas';

export default function* rootSaga() {
  yield all([
    ...userSagas,
  ]);
}
