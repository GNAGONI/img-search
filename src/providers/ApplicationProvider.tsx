
import Spinner from '../modules/Spinner';
import Notification from '../modules/Notification';
import StateProvider from './StateProvider';
import RouterProvider from './RouterProvider';
import ThemeProvider from './ThemeProvider';

const AppProvider = () => (
  <StateProvider>
    <ThemeProvider>
      <Spinner />
      <Notification />
      <RouterProvider />
    </ThemeProvider>
  </StateProvider>
);

export default AppProvider;
