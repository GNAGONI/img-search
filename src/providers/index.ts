import ApplicationProvider from './ApplicationProvider';
import StateProvider from './StateProvider';

export { 
  ApplicationProvider,
  StateProvider,
};