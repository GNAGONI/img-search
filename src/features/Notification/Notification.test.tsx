import renderer from 'react-test-renderer';
import Notification from './index';
import { StateProvider } from '../../providers';

it('renders correctly Notification', () => {
  const tree = renderer
    .create(
      <StateProvider>
        <Notification />
      </StateProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
