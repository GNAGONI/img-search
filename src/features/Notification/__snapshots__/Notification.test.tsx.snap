// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Notification 1`] = `
<div
  className="css-f779bt-MuiStack-root"
/>
`;
